testhelpers/testserver/fake_node.go

```diff
@@ -0,0 +1,324 @@
+	return &pb.LayerDurationResponse{Duration: &pb.SimpleInt{Value: m.seed.LayersDuration}}, nil
+}
+
+func (m meshServiceWrapper) MaxTransactionsPerSecond(context.Context, *pb.MaxTransactionsPerSecondRequest) (*pb.MaxTransactionsPerSecondResponse, error) {
+	return &pb.MaxTransactionsPerSecondResponse{MaxTxsPerSecond: &pb.SimpleInt{Value: m.seed.MaxTransactionPerSecond}}, nil
+}
+
+func (d *debugServiceWrapper) Accounts(context.Context, *empty.Empty) (*pb.AccountsResponse, error) {
+	accs := make([]*pb.Account, 0, len(d.seedGen.Accounts))
+	for _, acc := range d.seedGen.Accounts {
+		accs = append(accs, &pb.Account{
+			AccountId:    &pb.AccountId{Address: mustParse(acc.Account.Address)},
+			StateCurrent: nil,
+			StateProjected: &pb.AccountState{
+				Balance: &pb.Amount{Value: acc.Account.Balance},
+				Counter: acc.Account.Counter,
+			},
+		})
+	}
+	return &pb.AccountsResponse{AccountWrapper: accs}, nil
+}
+
+func (s *smesherServiceWrapper) PostConfig(context.Context, *empty.Empty) (*pb.PostConfigResponse, error) {
+	return &pb.PostConfigResponse{
+		BitsPerLabel:  s.seed.BitsPerLabel,
+		LabelsPerUnit: s.seed.LabelsPerUnit,
+		MinNumUnits:   s.seed.MinNumUnits,
+		MaxNumUnits:   s.seed.MaxNumUnits,
+	}, nil
+}
+
+func (g *globalStateServiceWrapper) GlobalStateStream(request *pb.GlobalStateStreamRequest, stream pb.GlobalStateService_GlobalStateStreamServer) error {
+	<-stateSynced
+	println("global state stream started")
+	for _, epoch := range g.seedGen.Epochs {
+		for _, reward := range epoch.Rewards {
+			resp := &pb.GlobalStateStreamResponse{Datum: &pb.GlobalStateData{Datum: &pb.GlobalStateData_Reward{
+				Reward: &pb.Reward{
+					LayerComputed: &pb.LayerNumber{Number: reward.LayerComputed},
+					Layer:         &pb.LayerNumber{Number: reward.Layer},
+					Total:         &pb.Amount{Value: reward.Total},
+					LayerReward:   &pb.Amount{Value: reward.LayerReward},
+					Coinbase:      &pb.AccountId{Address: mustParse(reward.Coinbase)},
+					Smesher:       &pb.SmesherId{Id: mustParse(reward.Smesher)},
+				},
+			}}}
+			if err := stream.Send(resp); err != nil {
+				return err
+			}
+		}
+	}
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			println("global state stream sending")
+		case <-stream.Context().Done():
+			return nil
+		}
+	}
+}
+
+func (g *globalStateServiceWrapper) Account(_ context.Context, req *pb.AccountRequest) (*pb.AccountResponse, error) {
+	accAddr := utils.BytesToAddressString(req.AccountId.Address)
+	acc, ok := g.seedGen.Accounts[strings.ToLower(accAddr)]
+	if !ok {
+		return nil, status.Error(codes.NotFound, "account not found")
+	}
+	return &pb.AccountResponse{
+		AccountWrapper: &pb.Account{
+			AccountId: &pb.AccountId{
+				Address: mustParse(accAddr),
+			},
+			StateCurrent: &pb.AccountState{
+				Balance: &pb.Amount{Value: acc.Account.Balance},
+				Counter: acc.Account.Counter,
+			},
+			StateProjected: nil,
```

Comment:
noticed a few places like this where `nil` is explicitly set when can be skipped